File: app/src/main/java/com/checkin/app/checkin/manager/fragments/ManagerInvoiceFragment.java
package com.checkin.app.checkin.manager.fragments;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.checkin.app.checkin.R;
import com.checkin.app.checkin.Shop.Private.Invoice.RestaurantSessionModel;
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceDetailActivity;
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceSessionAdapter;
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceViewModel;
import com.checkin.app.checkin.data.resource.Resource;
import com.checkin.app.checkin.misc.fragments.BaseFragment;
import com.checkin.app.checkin.utility.Utils;

import butterknife.BindView;

public class ManagerInvoiceFragment extends BaseFragment implements ShopInvoiceSessionAdapter.ShopInvoiceInteraction {

    @BindView(R.id.rv_shop_invoice_sessions)
    RecyclerView rvSessions;

    private ShopInvoiceSessionAdapter mAdapter;
    private ShopInvoiceViewModel mViewModel;

    public ManagerInvoiceFragment() {
    }

    public static ManagerInvoiceFragment newInstance() {
        return new ManagerInvoiceFragment();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        setupUi();
        mViewModel = ViewModelProviders.of(requireActivity()).get(ShopInvoiceViewModel.class);

        mViewModel.getRestaurantSessions().observe(this, input -> {
            if (input == null)
                return;
            if (input.getStatus() == Resource.Status.SUCCESS && input.getData() != null) {
                mAdapter.setSessionData(input.getData());
            } else if (input.getStatus() != Resource.Status.LOADING) {
                Utils.toast(requireContext(), input.getMessage());
            }
        });
    }

    private void setupUi() {
        rvSessions.setLayoutManager(new LinearLayoutManager(requireContext(), RecyclerView.VERTICAL, false));
        mAdapter = new ShopInvoiceSessionAdapter(this);
        rvSessions.setAdapter(mAdapter);
    }

    @Override
    protected int getRootLayout() {
        return R.layout.fragment_manager_invoice;
    }

    @Override
    public void onClickSession(RestaurantSessionModel data) {
        Intent intent = new Intent(requireContext(), ShopInvoiceDetailActivity.class);
        intent.putExtra(ShopInvoiceDetailActivity.KEY_SESSION_DATA, data);
        startActivity(intent);
    }
}

File: app/src/main/java/com/checkin/app/checkin/manager/fragments/ManagerInvoiceFragment.kt
package com.checkin.app.checkin.manager.fragments

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.fragment.app.activityViewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import com.checkin.app.checkin.R
import com.checkin.app.checkin.Shop.Private.Invoice.RestaurantSessionModel
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceDetailActivity
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceSessionAdapter
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceSessionAdapter.ShopInvoiceInteraction
import com.checkin.app.checkin.Shop.Private.Invoice.ShopInvoiceViewModel
import com.checkin.app.checkin.data.resource.Resource
import com.checkin.app.checkin.misc.fragments.BaseFragment
import com.checkin.app.checkin.utility.Utils

class ManagerInvoiceFragment : BaseFragment(), ShopInvoiceInteraction {
    override val rootLayout = R.layout.fragment_manager_invoice

    @BindView(R.id.rv_shop_invoice_sessions)
    internal lateinit var rvSessions: RecyclerView

    private val mAdapter: ShopInvoiceSessionAdapter by lazy {
        ShopInvoiceSessionAdapter(this)
    }
    private val mViewModel: ShopInvoiceViewModel by activityViewModels()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        initRefreshScreen(R.id.sr_manager_invoice)
        setupUi()
        mViewModel.restaurantSessions.observe(viewLifecycleOwner, Observer {
            it?.let { input ->
                handleLoadingRefresh(input)
                if (input.status === Resource.Status.SUCCESS && input.data != null) {
                    mAdapter.setSessionData(input.data)
                }
            }
        })
    }

    private fun setupUi() {
        rvSessions.layoutManager = LinearLayoutManager(requireContext(), RecyclerView.VERTICAL, false)
        rvSessions.adapter = mAdapter
    }

    override fun updateScreen() {
        super.updateScreen()
        Utils.getCurrentFormattedDateInvoice().also { currDate ->
            mViewModel.filterFrom(currDate, doFetch = false)
            mViewModel.filterTo(currDate, doFetch = false)
        }
        mViewModel.fetchShopSessions()
    }

    override fun onClickSession(data: RestaurantSessionModel?) {
        context?.let { ctx ->
            Intent(ctx, ShopInvoiceDetailActivity::class.java).apply {
                putExtra(ShopInvoiceDetailActivity.KEY_SESSION_DATA, data)
            }.also { startActivity(it) }
        }
    }

    companion object {
        fun newInstance() = ManagerInvoiceFragment()
    }
}
